integrated_channels/integrated_channel/learner_data.py

```diff
@@ -134,6 +133,52 @@ def collect_learner_data(self):
                 is_passing=is_passing,
             )
 
+    def _should_skip_enrollments_data_collection(self, enterprise_enrollment):
+        """
+        Specifies whether to skip data collection of the enterprise enrollment due to any configuration reason.
+
+        Reasons for skipping data collection may include data sharing consent not being granted,
+        or audit track data reporting being disabled for audit enrollments, etc..
+
+        :param enterprise_enrollment: The enterprise enrollment whose configuration is to be checked against.
+        :return: True if we should skip, False otherwise.
+        """
+        return any([
+            not enterprise_enrollment.consent_available,
+            self._disabled_audit_reporting(enterprise_enrollment),
+        ])
+
+    def _disabled_audit_reporting(self, enterprise_enrollment):
```

Comment:
I've thought about making it a property, but it makes an API call, so not sure how much I want to couple the course enrollment API with the EnterpriseCourseEnrollment model.

I will rename it though!